NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+
+    ForkThread(function()
+        Cinematics.EnterNISMode()
+        Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_03a'), 1)
+        WaitSeconds(1)
+        Cinematics.CameraTrackEntity(ScenarioInfo.EnemyCommander, 40, 2)
+        WaitSeconds(5)
+        Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_03b'), 1)
+        WaitSeconds(5)
+        VisMarker3_1:Destroy()
+        VisMarker3_2:Destroy()
+        ScenarioFramework.ClearIntel(ScenarioUtils.MarkerToPosition('M3_UEF_Base_Marker'), 80)
+        ScenarioFramework.ClearIntel(ScenarioUtils.MarkerToPosition('M3PBase'), 40)
+        Cinematics:ExitNISMode()
+    end)
+
+    ----------
+    -- Unlock T2 Air Factory
+    ----------
+    ScenarioFramework.PlayUnlockDialogue()
+    for _, player in ScenarioInfo.HumanPlayers do
+        ScenarioFramework.RemoveRestriction(player, categories.inb0202)
+        ScenarioFramework.RemoveRestriction(player, categories.inb0212)
+    end
+
+    ----------
+    -- Objectives
+    ----------
+    ScenarioInfo.M3P1 = Objectives.KillOrCapture(
+    'primary',                      -- type
+    'incomplete',                   -- complete
+    'Destroy Enemy Commander',  -- title
+    'The Enemy Commander is in sight and we need to kill him. Use all available units to destroy him.', -- description
+        {                               -- target
+            Units = {ScenarioInfo.EnemyCommander}
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M3P1)
+
+    ScenarioInfo.M3S1 = Objectives.CategoriesInArea(
+    'secondary',                      -- type
+    'incomplete',                   -- complete
+    'Destroy Enemy Power Generators',  -- title
+    'We believe this Power Station is powering the shields in the enemy base. Destroy them.',  -- description
+    'kill',
+    {
+        MarkUnits = true,
+        Requirements = {
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+            { Area = 'M3Power', Category = categories.ueb1201, CompareOp = '<=', Value = 0, ArmyIndex = UEF},
+        },
+    })
+
+    table.insert(AssignedObjectives, ScenarioInfo.M3S1)
+
+    ScenarioInfo.M3S2 = Objectives.CategoriesInArea(
+    'secondary',                      -- type
+    'incomplete',                   -- complete
+    'Construct Tech 2 Air Factory',  -- title
+    'Our Tech 2 Air Factories provide the ability to call in orbital strikes. Construct one and use the ability.',  -- description
+    'Build',
+    {
+        MarkUnits = false,
+        MarkArea = false,
+        Requirements =
+        {
+            {Area = 'M3', Category = categories.inb0202, CompareOp = '>=', Value = 1, ArmyIndex = Player1},
+        },
+    })
+
+    table.insert(AssignedObjectives, ScenarioInfo.M3S2)
+
+    ScenarioInfo.M3P1:AddResultCallback(

```

Comment:
you should put this at line 615 right after you create the objective, to keep things together, same with the other two result callbacks